### ISSUE TITLE
Export doc (at least line groups) in usable format

### ISSUE THREAD

**Issue by shreevatsa** [shreevatsa/ambuda#46]
Now that marking into groups (#43) is working, I think I already have most of what's needed for a (bare minimum) "Bookchop" (#32) MVP for personal use. The only thing to add is some way to use the manually marked groups outside of the proofreading UI. This is probably simple—I'm already saving the PM doc as JSON to the db, so probably just need to get out the doc and parse it—but creating this tracking issue as there may be things to think about.

**Comment by shreevatsa** [comment]
Example of what I'd like: I locally have this page http://localhost:5000/proofing/epigramsbhartrhari/128/ done:

<img width="883" alt="image" src="https://user-images.githubusercontent.com/767530/222987779-1466644e-89db-440e-a32e-8691fdcfff82.png">

From the db, I'd like to get out a map like:

50: (region)

51: (region)

53: (regions)

etc.

A map from each name to all the regions (on all pages) with that name.

Or maybe we can start with extracting this for one page.

**Comment by shreevatsa** [comment]
```
sqlite3 deploy/data/database/database.db
```

**Comment by shreevatsa** [comment]
select * from proof_revisions where page_id = 128

**Comment by shreevatsa** [comment]
Per https://stackoverflow.com/a/48328243 the sqlite way of getting value with max is just max:

    select content, max(created) from proof_revisions where page_id = 134

**Comment by shreevatsa** [comment]
I'll have to change this query when I have more proofing projects, but this works for now:

```
sqlite3 deploy/data/database/database.db "select content from (select content, max(created) from proof_revisions where page_id = 128)" | jq
```

**Comment by shreevatsa** [comment]
` jq '.content[] | select(.attrs.groupName=="50")'` to get all the lines:

```
{
  "type": "lgVerse",
  "attrs": {
    "groupName": "50"
  },
  "content": [
    {
      "type": "line",
      "attrs": {
        "box": {
          "xmin": 580,
          "xmax": 2355,
          "ymin": 440,
          "ymax": 623
        }
      },
      "content": [
        {
          "type": "text",
          "text": "दानं भोगो नाशस् तिस्रो गतयो भवन्ति वित्तस्य ।"
        }
      ]
    },
    {
      "type": "line",
      "attrs": {
        "box": {
          "xmin": 576,
          "xmax": 2479,
          "ymin": 612,
          "ymax": 769
        }
      },
      "content": [
        {
          "type": "text",
          "text": "यो न ददाति न भुते तस्य तृतीया गतिर्भवति ॥ ५० ॥"
        }
      ]
    }
  ]
}
```

**Comment by shreevatsa** [comment]
```
jq '.content[] | select(.attrs.groupName=="50") | .content[].attrs'
{
  "box": {
    "xmin": 580,
    "xmax": 2355,
    "ymin": 440,
    "ymax": 623
  }
}
{
  "box": {
    "xmin": 576,
    "xmax": 2479,
    "ymin": 612,
    "ymax": 769
  }
}
```

**Comment by shreevatsa** [comment]
```
 jq '.content[] | select(.attrs.groupName=="50") | [.content[].attrs.box.xmin] | min' gives 576
 jq '.content[] | select(.attrs.groupName=="50") | [.content[].attrs.box.xmax] | max' gives 2479
 jq '.content[] | select(.attrs.groupName=="50") | [.content[].attrs.box.ymin] | min' gives 440
 jq '.content[] | select(.attrs.groupName=="50") | [.content[].attrs.box.ymax] | max' gives 769
```

**Comment by shreevatsa** [comment]
Could loop over pages and do this for each… wondering if I should just do the whole thing in Python (or [Deno](https://matklad.github.io/2023/02/12/a-love-letter-to-deno.html)).

**Comment by shreevatsa** [comment]
Back. I think 

    select * from proof_pages;

gives the actual ids of the pages, and can try 

    select * from proof_pages where version > 0;

to see which `version` to use for each page. For example:

```
sqlite> select * from proof_pages where version > 0;
┌─────┬────────────┬──────┬───────┬─────────┬────────────────────┬───────────┐
│ id  │ project_id │ slug │ order │ version │ ocr_bounding_boxes │ status_id │
├─────┼────────────┼──────┼───────┼─────────┼────────────────────┼───────────┤
│ 128 │ 1          │ 128  │ 128   │ 1       │                    │ 2         │
│ 130 │ 1          │ 130  │ 130   │ 5       │                    │ 2         │
│ 131 │ 1          │ 131  │ 131   │ 1       │                    │ 2         │
│ 133 │ 1          │ 133  │ 133   │ 1       │                    │ 2         │
│ 134 │ 1          │ 134  │ 134   │ 2       │                    │ 2         │
│ 200 │ 1          │ 200  │ 200   │ 2       │                    │ 2         │
└─────┴────────────┴──────┴───────┴─────────┴────────────────────┴───────────┘
```

means that for page 134, should use version 2. But that's not helpful AFAICT?

```
select id, project_id, page_id, author_id, status_id, created, summary from proof_revisions where page_id=134;
┌────┬────────────┬─────────┬───────────┬───────────┬────────────────────────────┬─────────────┐
│ id │ project_id │ page_id │ author_id │ status_id │          created           │   summary   │
├────┼────────────┼─────────┼───────────┼───────────┼────────────────────────────┼─────────────┤
│ 5  │ 1          │ 134     │ 2         │ 2         │ 2023-02-26 15:44:44.501384 │ saving json │
│ 6  │ 1          │ 134     │ 2         │ 2         │ 2023-02-26 15:48:28.400609 │ saving      │
└────┴────────────┴─────────┴───────────┴───────────┴────────────────────────────┴─────────────┘
```

so the `version` column is only for handling edit conflicts I guess.

**Comment by shreevatsa** [comment]
But at least the `page_id` is there.

    select id from proof_pages where project_id = 1;

and loop only over those ids.

**Comment by shreevatsa** [comment]
Added a script in f354120d6b6890cbc260313d045eea92ec36d570:

```
python3 get_proof_lg_regions.py
50 [{'xmin': 576, 'xmax': 2479, 'ymin': 440, 'ymax': 769}]
51 [{'xmin': 573, 'xmax': 2615, 'ymin': 769, 'ymax': 1063.5}]
52 [{'xmin': 568, 'xmax': 2619, 'ymin': 1063.5, 'ymax': 1395.5}]
53 [{'xmin': 567, 'xmax': 2275, 'ymin': 1395.5, 'ymax': 1723.5}]
50f [{'xmin': 282, 'xmax': 2619, 'ymin': 1723.5, 'ymax': 2667}]
51f [{'xmin': 282, 'xmax': 2622, 'ymin': 2667, 'ymax': 3500.5}]
52f [{'xmin': 278, 'xmax': 2609, 'ymin': 3500.5, 'ymax': 4188.5}]
53f [{'xmin': 276, 'xmax': 2607, 'ymin': 4188.5, 'ymax': 4678}]
```

Hmm, it needs page numbers…

**Comment by shreevatsa** [comment]
Fixed in 46138ffdd9cff2ced3e9bcb561571eb6319411cf:

```
python3 get_proof_lg_regions.py
50 [{'page_id': 128, 'xmin': 576, 'xmax': 2479, 'ymin': 440, 'ymax': 769}]
51 [{'page_id': 128, 'xmin': 573, 'xmax': 2615, 'ymin': 769, 'ymax': 1063.5}]
52 [{'page_id': 128, 'xmin': 568, 'xmax': 2619, 'ymin': 1063.5, 'ymax': 1395.5}]
53 [{'page_id': 128, 'xmin': 567, 'xmax': 2275, 'ymin': 1395.5, 'ymax': 1723.5}]
50f [{'page_id': 128, 'xmin': 282, 'xmax': 2619, 'ymin': 1723.5, 'ymax': 2667}]
51f [{'page_id': 128, 'xmin': 282, 'xmax': 2622, 'ymin': 2667, 'ymax': 3500.5}]
52f [{'page_id': 128, 'xmin': 278, 'xmax': 2609, 'ymin': 3500.5, 'ymax': 4188.5}]
53f [{'page_id': 128, 'xmin': 276, 'xmax': 2607, 'ymin': 4188.5, 'ymax': 4678}]
```

**Comment by shreevatsa** [comment]
Did the next page that also contains 53 and 53f:

<img width="590" alt="image" src="https://user-images.githubusercontent.com/767530/222995303-dd32a94a-c0ef-417e-a622-409bf10ca58e.png">

and now the script gives:

```
 python3 get_proof_lg_regions.py
50 [{'page_id': 128, 'xmin': 576, 'xmax': 2479, 'ymin': 440, 'ymax': 769}]
51 [{'page_id': 128, 'xmin': 573, 'xmax': 2615, 'ymin': 769, 'ymax': 1063.5}]
52 [{'page_id': 128, 'xmin': 568, 'xmax': 2619, 'ymin': 1063.5, 'ymax': 1395.5}]
53 [{'page_id': 128, 'xmin': 567, 'xmax': 2275, 'ymin': 1395.5, 'ymax': 1723.5}, {'page_id': 129, 'xmin': 581, 'xmax': 2611, 'ymin': 427, 'ymax': 786.5}]
50f [{'page_id': 128, 'xmin': 282, 'xmax': 2619, 'ymin': 1723.5, 'ymax': 2667}]
51f [{'page_id': 128, 'xmin': 282, 'xmax': 2622, 'ymin': 2667, 'ymax': 3500.5}]
52f [{'page_id': 128, 'xmin': 278, 'xmax': 2609, 'ymin': 3500.5, 'ymax': 4188.5}]
53f [{'page_id': 128, 'xmin': 276, 'xmax': 2607, 'ymin': 4188.5, 'ymax': 4678}, {'page_id': 129, 'xmin': 282, 'xmax': 2619, 'ymin': 1815, 'ymax': 2623.5}]
54 [{'page_id': 129, 'xmin': 564, 'xmax': 2615, 'ymin': 786.5, 'ymax': 1113.5}]
55 [{'page_id': 129, 'xmin': 704, 'xmax': 2434, 'ymin': 1113.5, 'ymax': 1449}]
56 [{'page_id': 129, 'xmin': 562, 'xmax': 2613, 'ymin': 1449, 'ymax': 1815}]
54f [{'page_id': 129, 'xmin': 276, 'xmax': 2620, 'ymin': 2623.5, 'ymax': 3565.5}]
55f [{'page_id': 129, 'xmin': 272, 'xmax': 2625, 'ymin': 3565.5, 'ymax': 4678}]
```

Great!

Maybe when we actually start trying to use these values we'll run into interesting problems, but this is good for now.
